### ISSUE TITLE
Confusing “Installer” in upgrade with Homebrew

### ISSUE THREAD

**Issue by yhoiseth** [apparition47/MailTrackerBlocker#106]
Hi,

Thanks a ton for MailTrackerBlocker.

I often run `brew update && brew outdated` followed by `brew upgrade` in order to keep things up to date. Whenever MailTrackerBlocker is outdated, running `brew upgrade` results in this “Installer” Finder window that I don’t understand. For example, today, `brew update && brew upgrade` showed

```
yngve → brew update && brew outdated
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
…
You have 6 outdated formulae and 1 outdated cask installed.
You can upgrade them with brew upgrade
or list them with brew outdated.
…
mailtrackerblocker (0.3.24) != 0.3.29
```

I ran `brew upgrade`, and I got this output:

```
…
==> Upgrading 1 outdated package:
mailtrackerblocker 0.3.24 -> 0.3.29
==> Upgrading mailtrackerblocker
==> Downloading https://github.com/apparition47/MailTrackerBlocker/releases/download/0.3.29/MailTrackerBlocker.pkg
==> Downloading from https://github-releases.githubusercontent.com/277057483/eb780a80-968d-11eb-8108-d1606a0a448b?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSV
######################################################################## 100.0%
==> Uninstalling packages; your password may be necessary:
com.onefatgiraffe.mailtrackerblocker
Password:
==> Removing files:
/Library/Mail/Bundles/MailTrackerBlocker.mailbundle
==> Running installer for mailtrackerblocker; your password may be necessary.
Package installers may write to any location; options such as `--appdir` are ignored.
installer: Package name is MailTrackerBlocker
installer: Installing at base path /
installer: The install was successful.
==> Purging files for version 0.3.24 of Cask mailtrackerblocker
🍺  mailtrackerblocker was successfully upgraded!
==> Caveats
…
```

After entering my password, I got this window:

<img width="1152" alt="Screenshot 2021-04-09 at 16 23 53" src="https://user-images.githubusercontent.com/8469540/114195960-37a6a180-9951-11eb-9d70-2bfa799642cb.png">

Whenever I get this window, I just click _Cancel_ because I don’t understand what file I am supposed to open. It doesn’t seem to cause any problems, but I don’t know.

**Comment by apparition47** [comment]
In order to auto-enable Big Sur, I need to copy the `.mailbundle` into a privileged user dir (specifically: `~/Library/Application\ Support/Mail/Plug-ins/Bundles/Library/Mail/Bundles/`). This isn't possible for the installer script to do normally. So I made an [Apple Script to automate Finder to copy it into that dir](https://github.com/apparition47/MailTrackerBlocker/blob/main/Packaging/postinstall#L64-L73) as a workaround.

You should've seen this macOS security prompt that comes up the first time the Apple Script is executed:
<img src="https://user-images.githubusercontent.com/3298414/113468821-8fda3100-9483-11eb-9fe8-9e55a189b27f.png" width="150">

If you've already authorized/OK'd the script in the past, the script will run successfully but a Finder window will be left open (like your screenshot) for some reason.

Confusing, yes 😅. I'll see if I can fix it.

**Comment by yhoiseth** [comment]
Thanks for the quick response.

If I understand you correctly, I can safely click _Cancel_ in the Finder window?

If the issue turns out to be hard to fix, maybe adding instructions as a [caveat](https://docs.brew.sh/Formula-Cookbook#caveats) could be an acceptable workaround?

**Comment by apparition47** [comment]
Ah yes, that's right!

>adding instructions as a caveat

The Homebrew team is very, very strict about what I can write in the Brew recipe as a caveat. I had to delete my original instructions there because they want developers to only have notes strictly related to installation. 😵

**Comment by yhoiseth** [comment]
> Ah yes, that's right!

Great :)

> The Homebrew team is very, very strict about what I can write in the Brew recipe as a caveat. I had to delete my original instructions there because they want developers to only have notes strictly related to installation. 😵

Oh, I didn’t know that. How unfortunate.

Have a great weekend :)

**Comment by apparition47** [comment]
To be resolved in #195. See #194 for details.
